components/lib/state/src/main/java/mozilla/components/lib/state/Store.kt

```diff
@@ -38,11 +25,23 @@ typealias Observer<S> = (S) -> Unit
  *
  * @param initialState The initial state until a dispatched [Action] creates a new state.
  * @param reducer A function that gets the current [State] and [Action] passed in and will return a new [State].
+ * @param middleware Optional list of [Middleware] sitting between the [Store] and the [Reducer].
  */
 open class Store<S : State, A : Action>(
     initialState: S,
-    private val reducer: Reducer<S, A>
+    reducer: Reducer<S, A>,
+    middleware: List<Middleware<S, A>> = emptyList()
 ) {
+    private val chain: (S, A) -> S by lazy {
```

Comment:
nit: what do you think about calling this function `reduce`. Chain introduces an additional name/concept and we're really only reducing the state here using the configured reducer and middleware.

Reply:
Yeah, that makes sense. Wasn't super happy about that name. :)